util/src/com/intellij/rt/coverage/instrumentation/ExtraFieldInstrumenter.java

```diff
@@ -132,6 +131,8 @@ protected MethodVisitor createMethodVisitor(final MethodVisitor mv,
    * Generate field with {@link ExtraFieldInstrumenter#myFieldType} array
    */
   public void generateMembers(ClassVisitor cv) {
```

Comment:
now the method would generate field even for interfaces in java 7 where it's impossible to have non-abstract methods. I assume, that this method should not be called if `myInterface == true`, but if that's the case, then it's strange to see the first if here. If not, then the sematics was changed

Reply:
You are right. (I changed this logic to avoid double field declaration.) I'll fix the conditions to add the field only when needed.